tests/end_to_end/features/steps/serving.py

```diff
@@ -192,16 +208,26 @@ def step_impl(context):
     )
 
 
-@then("I can get a list containing the entity name and a tuple with all the features")
-def step_impl(context):
+@then(
+    'I can get a list containing the entity name and a tuple with all the features from "{provider}"'
+)
+def step_impl(context, provider):
     i = 0
     for entity, features in context.iter:
-        if i >= len(context.expected):
-            break
-        print(entity, features)
-        assert entity == context.expected[i][0]
-        assert Counter(features) == Counter(context.expected[i][1])
-        i += 1
+        if provider == "snowflake":
+            if i >= len(context.expected):
+                break
+            print(entity, features)
+            assert entity == context.expected[i][0]
+            assert Counter(features) == Counter(context.expected[i][1])
+            i += 1
+        elif provider == "spark":
+            if i >= len(context.expected):
+                break
+            assert entity in context.expected
```

Comment:
why these different?

Reply:
I opted for a dictionary to avoid the order mattering, hence the difference.